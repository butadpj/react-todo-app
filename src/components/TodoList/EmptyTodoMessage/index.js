import React from 'react';

const EmptyTodoMessage = () => {
  return (
    <p className='empty-todo-message' style={{ textAlign: 'center', padding: '0 1.5rem' }}>
      Seems quiet here... Add your first todo.
    </p>
  );
};

export default EmptyTodoMessage
