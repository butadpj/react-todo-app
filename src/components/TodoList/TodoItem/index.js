import React from 'react';

const TodoItem = (props) => {
  const todoNameBasedOnState = (todo) => {
    return todo.isDone ? (
      <p className='isDone'>{`${todo.name} âœ…`}</p>
    ) : (
      <p>{todo.name}</p>
    );
  };

  return (
    <li>
      {todoNameBasedOnState(props.todo)}
      <div className='todo__buttons'>
        <button
          className={`btn btn--primary toggle-btn ${
            props.todo.isDone ? 'undo-btn' : 'done-btn'
          }`}
          onClick={() => props.toggleIsDone(props.todo)}
        >
          {props.todo.isDone ? 'Undo' : 'Done'}
        </button>
        <button
          className='btn delete-btn'
          onClick={() => props.deleteTodo(props.todo.id)}
        >
          Delete
        </button>
      </div>
    </li>
  );
};

export default TodoItem