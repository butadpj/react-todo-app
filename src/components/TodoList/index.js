import React from 'react';
import EmptyTodoMessage from './EmptyTodoMessage';
import TodoItem from './TodoItem';

const TodoList = ({todos = [], toggleIsDone, deleteTodo}) => {
  return (
    <ul className='todo__list'>
      {todos.length <= 0 ? (
        <EmptyTodoMessage />
      ) : (
        todos.map(todo => {
          return (
            <TodoItem
              key={todo.id}
              todo={todo} 
              toggleIsDone={toggleIsDone}
              deleteTodo={deleteTodo}
            />
          )
        })
      )}
    </ul>
  );
};

export default TodoList;
