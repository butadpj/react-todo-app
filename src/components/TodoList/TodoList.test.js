import TodoList from "./index";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

const todoList = [
  { id: 1, name: 'hey', isDone: false },
  { id: 2, name: 'jude', isDone: false },
  { id: 3, name: 'dont', isDone: false },
  { id: 4, name: 'make', isDone: false },
  { id: 5, name: 'it bad', isDone: true },
];

describe('<TodoList />', () => {
  it("renders without crashing", () => {
    shallow(<TodoList />);
  });

  it("UI doesn't change unexpectedly", () => {
    const withTodos = mount(<TodoList todos={todoList}/>);
    expect(toJson(withTodos)).toMatchSnapshot();

    const noTodos = mount(<TodoList />);
    expect(toJson(noTodos)).toMatchSnapshot();
  });

  it("Display <EmptyTodoMessage /> when there's no todos", () => {
    const undefinedTodos = mount(<TodoList />);
    expect(undefinedTodos.find('EmptyTodoMessage').exists()).toBe(true);

    const emptyTodos = mount(<TodoList todos={[]}/>);
    expect(emptyTodos.find('EmptyTodoMessage').exists()).toBe(true);
  });

  it("Render <TodoItem /> for each item in the list", () => {
    const wrapper = mount(<TodoList todos={todoList}/>);
    expect(wrapper.find('TodoItem').exists()).toBe(true);
    expect(wrapper.find('TodoItem')).toHaveLength(todoList.length);
  });
});