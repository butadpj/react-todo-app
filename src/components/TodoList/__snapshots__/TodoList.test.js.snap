// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> UI doesn't change unexpectedly 1`] = `
<TodoList
  todos={
    Array [
      Object {
        "id": 1,
        "isDone": false,
        "name": "hey",
      },
      Object {
        "id": 2,
        "isDone": false,
        "name": "jude",
      },
      Object {
        "id": 3,
        "isDone": false,
        "name": "dont",
      },
      Object {
        "id": 4,
        "isDone": false,
        "name": "make",
      },
      Object {
        "id": 5,
        "isDone": true,
        "name": "it bad",
      },
    ]
  }
>
  <ul
    className="todo__list"
  >
    <TodoItem
      key="1"
      todo={
        Object {
          "id": 1,
          "isDone": false,
          "name": "hey",
        }
      }
    >
      <li>
        <p>
          hey
        </p>
        <div
          className="todo__buttons"
        >
          <button
            className="btn btn--primary toggle-btn done-btn"
            onClick={[Function]}
          >
            Done
          </button>
          <button
            className="btn delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </li>
    </TodoItem>
    <TodoItem
      key="2"
      todo={
        Object {
          "id": 2,
          "isDone": false,
          "name": "jude",
        }
      }
    >
      <li>
        <p>
          jude
        </p>
        <div
          className="todo__buttons"
        >
          <button
            className="btn btn--primary toggle-btn done-btn"
            onClick={[Function]}
          >
            Done
          </button>
          <button
            className="btn delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </li>
    </TodoItem>
    <TodoItem
      key="3"
      todo={
        Object {
          "id": 3,
          "isDone": false,
          "name": "dont",
        }
      }
    >
      <li>
        <p>
          dont
        </p>
        <div
          className="todo__buttons"
        >
          <button
            className="btn btn--primary toggle-btn done-btn"
            onClick={[Function]}
          >
            Done
          </button>
          <button
            className="btn delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </li>
    </TodoItem>
    <TodoItem
      key="4"
      todo={
        Object {
          "id": 4,
          "isDone": false,
          "name": "make",
        }
      }
    >
      <li>
        <p>
          make
        </p>
        <div
          className="todo__buttons"
        >
          <button
            className="btn btn--primary toggle-btn done-btn"
            onClick={[Function]}
          >
            Done
          </button>
          <button
            className="btn delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </li>
    </TodoItem>
    <TodoItem
      key="5"
      todo={
        Object {
          "id": 5,
          "isDone": true,
          "name": "it bad",
        }
      }
    >
      <li>
        <p
          className="isDone"
        >
          it bad âœ…
        </p>
        <div
          className="todo__buttons"
        >
          <button
            className="btn btn--primary toggle-btn undo-btn"
            onClick={[Function]}
          >
            Undo
          </button>
          <button
            className="btn delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
        </div>
      </li>
    </TodoItem>
  </ul>
</TodoList>
`;

exports[`<TodoList /> UI doesn't change unexpectedly 2`] = `
<TodoList>
  <ul
    className="todo__list"
  >
    <EmptyTodoMessage>
      <p
        className="empty-todo-message"
        style={
          Object {
            "padding": "0 1.5rem",
            "textAlign": "center",
          }
        }
      >
        Seems quiet here... Add your first todo.
      </p>
    </EmptyTodoMessage>
  </ul>
</TodoList>
`;
